runtime/compiler/codegen/J9CodeGenerator.cpp

```diff
@@ -652,11 +652,12 @@ J9::CodeGenerator::lowerTreesPreChildrenVisit(TR::Node *parent, TR::TreeTop *tre
       {
       // J9
       //
-      // Hiding compressedref logic from CodeGen doesn't seem a good practise, the evaluator always need the uncompressedref node for write barrier,
-      // therefore, this part is deprecated. It'll be removed once P and Z update their corresponding evaluators.
+      // Hiding compressedref logic from CodeGen isn't a good practice, and the evaluator still needs the uncompressedref node for write barriers.
+      // Therefore, this part is deprecated. It can only be activated on X, P or Z with the TR_UseOldCompareAndSwapObject envvar.
       static bool UseOldCompareAndSwapObject = (bool)feGetEnv("TR_UseOldCompareAndSwapObject");
       static bool disableCASInlining = feGetEnv("TR_DisableCASInlining") != NULL;
-      if (self()->comp()->useCompressedPointers() && ((UseOldCompareAndSwapObject && (self()->comp()->target().cpu.isARM64() || !disableCASInlining)) || !(self()->comp()->target().cpu.isX86() || self()->comp()->target().cpu.isARM64())))
+      if (((self()->comp()->target().cpu.isX86() && !disableCASInlining) || self()->comp()->target().cpu.isPower() || self()->comp()->target().cpu.isZ()) &&
```

Comment:
Hmm, bit condfusing, `disableCASInlining` is only checked on x86 not on P and Z. If this is intended, We should have a comment.

Reply:
This is intended. The `TR_DisableCASInlining` envvar to disable inlining of both Unsafe `compareAndSwap` and `compareAndExchange` only has support on X. That's why the check is only on X.

Turning on `TR_DisableCASInlining` and `TR_UseOldCompareAndSwapObject` at the same time doesn't actually make sense since if CAS inlining is disabled, it won't use either of the compare and swap implementations. But, without this change it would crash and I didn't want that.

I will add a comment explaining that the change is because the `TR_DisableCASInlining` will take priority and the check is needed to make sure it doesn't crash.